source/hands_on/sql.rst

```diff
@@ -0,0 +1,173 @@
+soft Q-iteration in an equivalent form as minimizing
+
+.. image:: images/SQL_sto_Q.png
+
+where :math:`q_{s_{t}}` and :math:`q_{a_{t}}` are positive over :math:`\mathrm{S}` and :math:`\mathrm{A}` respectively.
+
+.. image:: images/SQL_sto_Q_tar.png
+
+is a target Q-value, with :math:`V^{\bar{\theta}}_{\text{soft}}` given by the equation 10 and :math:`\theta` being replaced by the target parameters, :math:`\bar{\theta}`.
+
+While the sampling distributions :math:`q_{s_{t}}`, :math:`q_{a_{t}}` and :math:`q_{a'}` can be arbitrary, we typically use real samples from rollouts of the current policy :math:`\pi(a_{t}|s_{t}) \propto exp(\frac{1}{\alpha} Q_{\text{soft}}^{\theta}(s_{t},a_{t}))`.
+
+However, in continuous spaces, Since the form of the policy is so
+general, sampling from it is intractable - We still need a tractable way to sample from the policy. Here is where SVGD comes in.
+
+Formally, we want to learn a state-conditioned stochastic neural network :math:`a_{t}=f^{\phi}(\xi,s_{t})` parametrized by :math:`\phi`, that
+maps noise samples :math:`\xi` drawn from a normal Gaussian, or other arbitrary distribution, into unbiased action samples. We denote
+the induced distribution of the actions as :math:`\pi^{\phi}(a_{t}|s_{t})` and we want tp find parameters :math:`\phi` so that the induced distribution approximates the energy-based distribution in terms of the
+KL divergence. 
+
+.. image:: images/SQL_sto_pi12.png
+
+Practically, we optimise the policy by the following two equations:
+
+.. image:: images/SQL_sto_pi13.png
+.. image:: images/SQL_sto_pi14.png
+
+Pseudo-code
+---------------
+The pseudo code is as follows:
+
+.. image:: images/SQL.png
+
+Where the equation 10, 11, 13, 14 can be referred from the above section.
+
+Extensions
+-----------
+SQL can be combined with:
+
+    - Exploration techniques such as epsilon-greedy or OU Noise (implemented in the original paper; Please refer to `Continuous control with deep reinforcement learning <https://arxiv.org/abs/1509.02971>`_ and `On the theory of the Brownian motion <https://link.aps.org/pdf/10.1103/PhysRev.36.823?casa_token=yFMSHBrxJoMAAAAA:5nFSMwUrqcdlUoobFDYOP0Y58r5jmNogkpHqFgMhzv0Md-4EcIkofMHHCkgsjEJFO10yqsmrhmNk_4dL>`_) to enchance explorations.
+
+    - Some analysts draw connection between Soft Q-learning and Policy
+      Gradient algorithms such as `Equivalence Between Policy Gradients and
+      Soft Q-Learning <https://arxiv.org/abs/1704.06440>`__.
+    
+    - SQL can be combined with demonstration data to propose an imitation learning algorithm: SQIL proposed in `SQIL: Imitation Learning via Reinforcement Learning with Sparse Rewards <https://arxiv.org/abs/1905.11108>`_.  Please also refer to `SQIL code <https://github.com/opendilab/DI-engine/blob/main/ding/policy/sqil.py>` for a DI-engine implementation.
+
```

Comment:
The link <https://github.com/opendilab/DI-engine/blob/main/ding/policy/sqil.py> is not available